go/libraries/doltcore/doltdb/commit_hooks.go

```diff
@@ -127,3 +184,69 @@ func (lh *LogHook) SetLogger(ctx context.Context, wr io.Writer) error {
 	lh.out = wr
 	return nil
 }
+
+func RunAsyncReplicationThreads(ctx context.Context, wg *sync.WaitGroup, ch chan PushArg, destDB *DoltDB, tmpDir string, logger io.Writer) {
+	mu := &sync.Mutex{}
+	var newHeads = make(map[string]PushArg, asyncPushBufferSize)
+
+	// stop first go routine before second
+	newCtx, stop := context.WithCancel(context.Background())
+	wg.Add(1)
+	go func() error {
```

Comment:
not actually doing anything with the error.  may want to at least make it clear that that is intentional

`_ = go func() error {...}()`